recipes/flann/all/test_package/test_package.cpp

```diff
@@ -0,0 +1,7 @@
+#include <flann/flann.hpp>
+
+int main()
+{
+    flann::Matrix<float> dataset;
+    flann::Matrix<float> query;
```

Comment:
These use default constructors of [templated classes](https://github.com/mariusmuja/flann/blob/1d04523268c388dabf1c0865d69e1b638c8c7d9d/src/cpp/flann/util/matrix.h#L108).
So no code from the packaged library is called and executed.
Maybe the examples at https://github.com/mariusmuja/flann/tree/master/examples are useful?

Reply:
They load datasets, which seemed like something you don't always want to ship with your packages because it's an additional 10Mo. Now if it isn't a problem I'm definitely adding that.

Reply:
10MiB is a bit too much, I'm afraid.
What could be done is skip the load, or create a dummy dataset in the same format as the example.

Reply:
I will try to come up with something when I find some time, maybe tomorrow.

Reply:
Each folder can consume only 256KB IIRC

Reply:
the point is just to verify example compiles and links correctly.
so, an actual example isn't needed. it should be enough to just call some API function (e.g. [flann_get_distance_type](https://github.com/mariusmuja/flann/blob/master/src/cpp/flann/flann.h#L106), or any other you prefer). you may also call the function that will fail, it's okay (it just shouldn't crash or hang at least).

Reply:
Apparently only the C bindings allow to test whether linking works easily since `flann_cpp.cpp` only includes `flann.hpp`. Therefore I had the recipe build the C bindings along with the rest, I guess that it's pretty harmless to have them too.